modules/data/meme_data.py

```diff
@@ -194,6 +194,140 @@ def insert_published_post(channel_message: Message):
+        """
+
+        report = DbManager.select_from(select="*",
+                                     table_name="user_report",
+                                     where="user_id = %s",
+                                     where_args=(user_id,))
+
+        if len(report) == 0:  # the vote is not present
+            return None
+        return report[0]
+
+    @staticmethod
+    def set_post_report(user_id: int, c_message_id: int, channel_id: int) -> bool:
+        """Adds the report of the user on a specific post
+
+        Args:
+            user_id (int): id of the user that reported
+            c_message_id (int): id of the post in question in the channel
+            channel_id (int): id of the channel
+
+        Returns:
+            bool: whether the report was added or removed
+        """
+        current_report = MemeData.get_post_report(user_id, c_message_id, channel_id)
+        report_added = True
+        if current_report is None:  # there isn't a report yet
+            DbManager.insert_into(table_name="spot_report",
+                                  columns=("user_id", "c_message_id", "channel_id"),
+                                  values=(user_id, c_message_id, channel_id))
+        else:  # the user has reported in past
+            report_added = False
+
+        return report_added
+
+    @staticmethod
+    def get_post_report(user_id: int, c_message_id: int, channel_id: int) -> Optional[str]:
+        """Gets the report of a specific user on a published post
+
+        Args:
+            user_id (int): id of the user that reported
+            c_message_id (int): id of the post in question in the channel
+            channel_id (int): id of the channel
+
+        Returns:
+            Optional[str]: value of the report or None if a report was not yet made
+        """
+
+        report = DbManager.select_from(select="*",
+                                     table_name="spot_report",
+                                     where="user_id = %s and c_message_id = %s and channel_id = %s",
+                                     where_args=(user_id, c_message_id, channel_id))
+
+        if len(report) == 0:  # the vote is not present
+            return None
+        return report[0]
+
+    @staticmethod
+    def get_last_post_report(user_id: int, channel_id: int) -> Optional[str]:
+        """Gets the last report of a specific user on a published post
+
+        Args:
+            user_id (int): id of the user that reported
+            c_message_id (int): id of the post in question in the channel
+            channel_id (int): id of the channel
+
+        Returns:
+            Optional[str]: last message id reported
+        """
+
+        report = DbManager.select_from(select="c_message_id",
+                                     table_name="spot_report",
+                                     where="user_id = %s and channel_id = %s",
+                                     where_args=(user_id, channel_id))
```

Comment:
Are you sure this will return the last report? Maybe an order by would be needed

Reply:
Coincidentally it is, because before send the new report, the last report will be removed :sweat_smile: . But now these methods will be removed after a soft rework.